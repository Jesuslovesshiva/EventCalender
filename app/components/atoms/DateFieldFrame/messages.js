import {defineMessages} from 'react-intl'

const scope = 'atoms.date-field-frame'

export default defineMessages({
  toggle: {
    id: `${scope}.toggle`,
    defaultMessage: 'Kalendar ein-/ausschalten',
    description: 'aria label toggle calendar button'
  }
})
