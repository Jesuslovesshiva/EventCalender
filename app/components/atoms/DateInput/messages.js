import {defineMessages} from 'react-intl'

const scope = 'atoms.date-picker'

export default defineMessages({
  ddmmyyyy: {
    id: `${scope}.date`,
    defaultMessage: 'TT.MM.JJJJ',
    description: 'placeholder for format dd.MM.yyyy'
  }
})
