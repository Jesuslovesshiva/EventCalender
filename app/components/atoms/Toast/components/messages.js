import {defineMessages} from 'react-intl'

const scope = 'atoms.toast'

export default defineMessages({
  success: {
    id: `${scope}.success`,
    defaultMessage: 'Info'
  },
  error: {
    id: `${scope}.error`,
    defaultMessage: 'Error or Warning'
  },
  alert: {
    id: `${scope}.alert`,
    defaultMessage: 'alert'
  },
  action: {
    id: `${scope}.action`,
    defaultMessage: 'Action'
  }
})
