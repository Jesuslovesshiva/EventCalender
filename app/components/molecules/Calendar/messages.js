import {defineMessages} from 'react-intl'

const scope = 'molecules.calendar'

export default defineMessages({
  selectMonth: {
    id: `${scope}.selectMonth`,
    defaultMessage: 'Monat auswählen'
  },
  selectYear: {
    id: `${scope}.selectYear`,
    defaultMessage: 'Jahr auswählen'
  },
  nextMonth: {
    id: `${scope}.nextMonth`,
    defaultMessage: 'nächstes Monat',
    description: 'aria label nextMonth button'
  },
  prevMonth: {
    id: `${scope}.prevMonth`,
    defaultMessage: 'vorheriges Monat',
    description: 'aria label prevMonth button'
  }
})
