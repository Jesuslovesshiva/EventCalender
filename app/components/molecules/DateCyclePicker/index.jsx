import {startOfWeek} from 'date-fns'
import FocusTrap from 'focus-trap-react'
import noop from 'lodash/noop'
import PropTypes from 'prop-types'
import {useCallback} from 'react'

import Popup from '@/components/atoms/Popup'
import useIsOpen from '@/hooks/use-is-open'
import usePopup from '@/hooks/use-popup'
import useSyncState from '@/hooks/use-sync-state'

import Field from './components/Field'
import Picker from './components/Picker'

const DateCyclePicker = props => {
  const [isOpen, {close, open, toggle}] = useIsOpen()
  const [value, setValue] = useSyncState(props.value)
  const {popper, referenceRef, popperRef} = usePopup({
    placement: 'bottom-start'
  })

  const onChange = useCallback(
    newDate => {
      const newValue = startOfWeek(newDate, {weekStartsOn: 1})
      props.onChange(newValue)
      setValue(newValue)
    },
    [props.onChange, setValue, props.value]
  )

  return (
    <FocusTrap
      active={isOpen}
      focusTrapOptions={{
        allowOutsideClick: true,
        initialFocus: false,
        onDeactivate: close,
        returnFocusOnDeactivate: false
      }}
    >
      <div>
        <Field
          ref={referenceRef}
          close={close}
          isOpen={isOpen}
          onChange={onChange}
          open={open}
          toggle={toggle}
          value={value}
        />

        <Popup
          // eslint-disable-next-line react/jsx-props-no-spreading
          {...popper.attributes.popper}
          ref={popperRef}
          style={popper.styles.popper}
          isOpen={isOpen}
        >
          <Picker
            close={close}
            isConfirmable={props.isConfirmable}
            onChange={onChange}
            onChangeMonth={props.onChangeMonth}
            value={value}
          />
        </Popup>
      </div>
    </FocusTrap>
  )
}

DateCyclePicker.propTypes = {
  isConfirmable: PropTypes.bool,
  onChange: PropTypes.func,
  onChangeMonth: PropTypes.func,
  value: PropTypes.instanceOf(Date)
}

DateCyclePicker.defaultProps = {
  isConfirmable: true,
  onChange: noop,
  onChangeMonth: noop,
  value: null
}

export default DateCyclePicker
