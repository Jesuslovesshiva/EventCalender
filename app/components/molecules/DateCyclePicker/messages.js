import {defineMessages} from 'react-intl'

const scope = 'molecules.date-cycle-picker'

export default defineMessages({
  from: {
    id: `${scope}.from`,
    defaultMessage: 'von',
    description: 'aria label from input'
  },
  to: {
    id: `${scope}.to`,
    defaultMessage: 'bis',
    description: 'aria label to input'
  },
  next: {
    id: `${scope}.next`,
    defaultMessage: 'n√§chste Woche',
    description: 'aria label next button'
  },
  prev: {
    id: `${scope}.prev`,
    defaultMessage: 'vorherige Woche',
    description: 'aria label prev button'
  }
})
