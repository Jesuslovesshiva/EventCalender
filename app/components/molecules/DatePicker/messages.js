import {defineMessages} from 'react-intl'

const scope = 'molecules.date-picker'

export default defineMessages({
  date: {
    id: `${scope}.date`,
    defaultMessage: 'Datum',
    description: 'aria label date input'
  }
})
