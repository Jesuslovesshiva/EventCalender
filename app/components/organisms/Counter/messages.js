import {defineMessages} from 'react-intl'

const scope = 'organisms.counter'

export default defineMessages({
  count: {
    id: `${scope}.count`,
    defaultMessage: 'Zähler: {count}',
    description: 'count output'
  },
  reset: {
    id: `${scope}.reset`,
    defaultMessage: 'Zurücksetzen',
    description: 'reset cta'
  }
})
