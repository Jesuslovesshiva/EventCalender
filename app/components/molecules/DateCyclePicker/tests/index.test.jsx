import {fireEvent, waitFor} from '@testing-library/react'

import dateFieldFrameMessages from '@/components/atoms/DateFieldFrame/messages'
import EventResultsInjector from '@/injectors/event/ResultsInjector'
import EVENT_TYPES from '@/modules/event/results/types'
import configureTestStore from '@/utils/test-utils/configure-test-store'
import render from '@/utils/test-utils/custom-render'
import renderInjectors from '@/utils/test-utils/render-injectors'
import setEntities from '@/utils/test-utils/set-entities'

import DateCyclePicker from '../index'
import messages from '../messages'

const getEventsForDate = ({date}) => [
  {
    id: 'event-1',
    startDate: date,
    title: 'the title',
    type: EVENT_TYPES.TRAINING
  }
]

const {clearDispatch, options, store} = configureTestStore()
const renderComponent = (props, renderFn = render) =>
  renderFn(
    <DateCyclePicker
      getEventsForDate={getEventsForDate}
      value={new Date(1)}
      {...props}
    />,
    options
  )

const event = {
  id: 'event-id-1',
  startedAt: new Date(2022, 3, 7).toISOString(),
  title: 'Tournament Title',
  type: EVENT_TYPES.TOURNAMENT
}

describe('DateCyclePicker', () => {
  beforeAll(() => {
    renderInjectors(<EventResultsInjector />, options)
    setEntities(store, {
      events: {[event.id]: event}
    })
  })

  beforeEach(clearDispatch)

  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should display range from start to end of week', () => {
    const {getByLabelText} = renderComponent({value: new Date(2022, 2, 8)})
    const from = getByLabelText(messages.from.defaultMessage)
    const to = getByLabelText(messages.to.defaultMessage)
    expect(from.value).toBe('07.03.2022')
    expect(to.value).toBe('13.03.2022')
  })

  it('should not break on invalid date', () => {
    const {getByLabelText} = renderComponent({value: new Date('foo')})
    const from = getByLabelText(messages.from.defaultMessage)
    const to = getByLabelText(messages.to.defaultMessage)
    expect(from.value).toBe('')
    expect(to.value).toBe('')
  })

  describe('events', () => {
    it('should call onChange prop on input change', () => {
      const onChange = jest.fn()
      const {getByLabelText} = renderComponent({onChange})
      const from = getByLabelText(messages.from.defaultMessage)
      fireEvent.change(from, {target: {value: '08.03.2022'}})
      fireEvent.blur(from)
      expect(onChange).toHaveBeenCalledWith(new Date(2022, 2, 7))
    })

    it('should call onChange prop on input change even if date stayed the same', () => {
      const onChange = jest.fn()
      const {getByLabelText} = renderComponent({
        onChange,
        value: new Date(2022, 2, 7)
      })
      const from = getByLabelText(messages.from.defaultMessage)
      fireEvent.change(from, {target: {value: '08.03.2022'}})
      fireEvent.blur(from)
      expect(onChange).toHaveBeenCalledWith(new Date(2022, 2, 7))
    })

    it('should change active state on focus', () => {
      const {container, getByLabelText} = renderComponent()
      const from = getByLabelText(messages.from.defaultMessage)
      fireEvent.focus(from)
      expect(container.firstChild.firstChild).toHaveClass('specific:bg-white')
    })

    it('should subtract one week on prev button click', () => {
      const onChange = jest.fn()
      const {getByLabelText} = renderComponent({
        onChange,
        value: new Date(2022, 2, 7)
      })
      const prev = getByLabelText(messages.prev.defaultMessage)
      fireEvent.click(prev)
      expect(onChange).toHaveBeenCalledWith(new Date(2022, 1, 28))
    })

    it('should add one week on next button click', () => {
      const onChange = jest.fn()
      const {getByLabelText} = renderComponent({
        onChange,
        value: new Date(2022, 2, 7)
      })
      const next = getByLabelText(messages.next.defaultMessage)
      fireEvent.click(next)
      expect(onChange).toHaveBeenCalledWith(new Date(2022, 2, 14))
    })

    it('should toggle picker on toggle button click', async () => {
      const {getByLabelText, queryByRole} = renderComponent()
      const toggle = getByLabelText(
        dateFieldFrameMessages.toggle.defaultMessage
      )
      fireEvent.click(toggle)
      expect(queryByRole('dialog')).toBeInTheDocument()
      fireEvent.click(toggle)
      await waitFor(() => expect(queryByRole('dialog')).not.toBeInTheDocument())
    })

    it('should close picker popup on Escape key down', async () => {
      const {container, getByLabelText, queryByRole} = renderComponent()
      const toggle = getByLabelText(
        dateFieldFrameMessages.toggle.defaultMessage
      )
      fireEvent.click(toggle)
      expect(queryByRole('dialog')).toBeInTheDocument()
      fireEvent.keyDown(container.firstChild, {key: 'Escape'})
      await waitFor(() => expect(queryByRole('dialog')).not.toBeInTheDocument())
    })
  })
})
