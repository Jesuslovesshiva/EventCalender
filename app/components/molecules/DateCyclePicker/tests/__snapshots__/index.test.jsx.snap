// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateCyclePicker should match the snapshot 1`] = `
<div>
  <div
    class="border border-gray-medium divide-x divide-gray-medium rounded inline-flex transition cursor-pointer hover:border-white hover:divide-white hover:bg-silver hover:shadow-md specific:divider-gray-light"
  >
    <button
      aria-label="vorherige Woche"
      class="w-[26px] transition-[border-color,color] text-gray-medium hover:text-gray-dark"
    >
      <div
        class="fill-current w-[1em] h-[1em] inline-block text-lg"
        data-icon="arrow_back_ios"
      />
    </button>
    <div
      class="flex items-center py-1 transition pl-1.5 pr-1.5"
    >
      <div
        class="fill-current w-[1em] h-[1em] inline-block text-lg mr-1"
        data-icon="date_range"
      />
      <input
        aria-label="von"
        class=" w-[75px] rounded-sm transition-[outline-color,background-color] bg-transparent font-bold text-center outline-transparent focus-visible:bg-silver focus-visible:outline focus-visible:outline-1 focus-visible:outline-gray-medium focus-visible:outline-offset-0 "
        placeholder="TT.MM.JJJJ"
        value="29.12.1969"
      />
      <div
        class="w-3.5 flex justify-center items-center font-bold"
      >
        -
      </div>
      <input
        aria-label="bis"
        class=" w-[75px] rounded-sm transition-[outline-color,background-color] bg-transparent font-bold text-center outline-transparent focus-visible:bg-silver focus-visible:outline focus-visible:outline-1 focus-visible:outline-gray-medium focus-visible:outline-offset-0 "
        placeholder="TT.MM.JJJJ"
        value="04.01.1970"
      />
      <button
        aria-label="Kalendar ein-/ausschalten"
        class="transition text-gray-medium hover:text-gray-dark"
      >
        <div
          class="fill-current w-[1em] h-[1em] inline-block text-2xl"
          data-icon="arrow_drop_down"
        />
      </button>
    </div>
    <button
      aria-label="nÃ¤chste Woche"
      class="w-[26px] transition-[border-color,color] text-gray-medium hover:text-gray-dark"
    >
      <div
        class="fill-current w-[1em] h-[1em] inline-block text-lg rotate-180"
        data-icon="arrow_back_ios"
      />
    </button>
  </div>
</div>
`;
